public/components/navigation.react.js

```diff
@@ -9,12 +9,19 @@ export default class NavigationComponent extends React.Component {
     }
 
     render () {
+        var distinctContentSources = [];
+        var allContentSources = this.props.data;
+        for(var i = allContentSources.length - 1; i >= 0; i--) {
+            if(distinctContentSources[allContentSources[i].id] == undefined)
+                distinctContentSources[allContentSources[i].id] = { appName: allContentSources[i].appName, environment: allContentSources[i].environment };
+        }
```

Comment:
I am not sure what this code exactly does, but could you not rewrite it in a more functional/scala way like:
```js
allContentSources.map( source => { source: source.appName, environment: environment: source.environment }).filter(function (value, index, self) {return self.indexOf(value) === index))
```